ci/check_wat.js

```diff
@@ -0,0 +1,54 @@
+const fs = require('fs').promises;
+const path = require('path');
+const wabtPromise = require('wabt')();
+
+async function processWatFile(filePath, wabt) {
+    console.log("Processing:", filePath);
+
+    const watString = await fs.readFile(filePath, 'utf8');
+
+    const wasmModule = wabt.parseWat(filePath, watString);
+    const binaryWasm = wasmModule.toBinary({}).buffer;
+
+    const importObject = {
+        env: {
+            assert_eq: (a, b) => {
+                if (a !== b) {
+                    throw new Error(`Assertion failed in ${filePath}: ${a} !== ${b}`);
+                }
+                console.log(`Assertion passed: ${a} === ${b}`);
+            }
+        }
+    };
+
+    await WebAssembly.instantiate(binaryWasm, importObject);
```

Comment:
Might be worth noting that for many (or even all?) processed WAT files `instantiate` is sufficient to run the modules `$main` function since these files have `(start $main)`.